internal/localizer/localizer.go

```diff
@@ -0,0 +1,109 @@
+package localizer
+
+import (
+	// Import the internal/translations so that it's init() function
+	// is run. It's really important that we do this here so that the
+	// default message catalog is updated to use our translations
+	// *before* we initialize the message.Printer instances below.
+
+	"fmt"
+	"os"
+
+	_ "github.com/microsoft/go-sqlcmd/internal/translations"
+	"golang.org/x/text/language"
+	"golang.org/x/text/message"
+)
+
+var Translator Localizer
+
+// Define a Localizer type which stores the relevant locale ID and a
+// (deliberately unexported) message.Printer instance for the locale.
+type Localizer struct {
+	ID      string
+	printer *message.Printer
+}
+
+// Initialize a slice which holds the initialized Localizer types for
+// each of our supported locales.
+var locales = []Localizer{
+	{
+		ID:      "de-de",
+		printer: message.NewPrinter(language.MustParse("de-DE")),
+	},
+	{
+		ID:      "fr-fr",
+		printer: message.NewPrinter(language.MustParse("fr-FR")),
+	},
+	{
+		ID:      "en-us",
+		printer: message.NewPrinter(language.MustParse("en-US")),
+	},
+	{
+		ID:      "zh-cn",
+		printer: message.NewPrinter(language.MustParse("zh-CN")),
+	},
+	{
+		ID:      "zh-tw",
+		printer: message.NewPrinter(language.MustParse("zh-TW")),
+	},
+	{
+		ID:      "it-it",
+		printer: message.NewPrinter(language.MustParse("it-IT")),
+	},
+	{
+		ID:      "ja-jp",
+		printer: message.NewPrinter(language.MustParse("ja-JP")),
+	},
+	{
+		ID:      "ko-kr",
+		printer: message.NewPrinter(language.MustParse("ko-KR")),
+	},
+	{
+		ID:      "pt-br",
+		printer: message.NewPrinter(language.MustParse("pt-BR")),
+	},
+	{
+		ID:      "ru-ru",
+		printer: message.NewPrinter(language.MustParse("ru-RU")),
+	},
+	{
+		ID:      "es-es",
+		printer: message.NewPrinter(language.MustParse("es-ES")),
```

Comment:
>NewPrinter(language.MustParse("es-ES")) [](http://example.com/codeflow?start=19&length=39)

is NewPrinter a func or a struct? If it's a struct, can we change  Localizer.printer to a func so it doesn't allocate memory until it's called? We should avoid creating a bunch of objects for all the languages when we're only going to use one of them

Reply:
I agree. No every language printer was needed to be initialized. I have refactored this code to instantiate only one language printer instance.